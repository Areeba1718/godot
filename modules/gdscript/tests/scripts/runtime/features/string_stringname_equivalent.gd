# https://github.com/godotengine/godot/issues/64171
# https://github.com/godotengine/godot/issues/60145

var s = "abc"
var sn = &"abc"

func test():
	print("Compare ==: ", "abc" == &"abc")
	print("Compare ==: ", &"abc" == "abc")
	print("Compare !=: ", "abc" != &"abc")
	print("Compare !=: ", &"abc" != "abc")

	print("Concat: ", "abc" + &"def")
	print("Concat: ", &"abc" + "def")
	print("Concat: ", &"abc" + &"def")

	print("literal StringName is String: ", &"abc" is String)
	print("StringName is String: ", sn is String)

	for c in sn:
		print(c)

	print(sn[1])
	print(sn[-1])

	match &"abc":
		"abc":
			print("String-like Match")
		_:
			print("no Match")

	match "abc":
		sn:
			print("String matched StringName")
		_:
			print("no match")

	match &"abc":
		s:
			print("StringName matched String")
		_:
			print("no match")
