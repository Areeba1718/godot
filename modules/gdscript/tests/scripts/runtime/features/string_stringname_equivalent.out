GDTEST_OK
Compare ==: true
Compare ==: true
Compare !=: false
Compare !=: false
Concat: abcdef
Concat: abcdef
Concat: abcdef
literal StringName is String: true
StringName is String: true
a
b
c
b
c
String-like Match
String matched StringName
StringName matched String
